ambry-cloud/src/main/java/com.github.ambry.cloud/azure/AzureCloudDestination.java

```diff
@@ -219,6 +213,44 @@ public boolean uploadBlob(BlobId blobId, long blobSize, CloudBlobMetadata cloudB
     }
   }
 
+  /**
+   * Upload the blob to Azure storage if it does not already exist in the designated container.
+   * @param blobId the blobId to upload
+   * @param inputLength the input stream length, if known (-1 if not)
+   * @param cloudBlobMetadata the blob metadata
+   * @param blobInputStream the input stream
+   * @return {@code true} if the upload was successful, {@code false} if the blob already exists.
+   * @throws StorageException
+   * @throws URISyntaxException
+   * @throws IOException
+   */
+  private boolean uploadIfNotExists(BlobId blobId, long inputLength, CloudBlobMetadata cloudBlobMetadata,
+      InputStream blobInputStream) throws StorageException, URISyntaxException, IOException {
+    BlobRequestOptions options = null; // may want to set BlobEncryptionPolicy here
+    AccessCondition condition = AccessCondition.generateIfNoneMatchCondition("*");
+    Timer.Context storageTimer = azureMetrics.blobUploadTime.time();
+    try {
+      CloudBlobContainer azureContainer = getContainer(blobId, true);
+      String azureBlobName = getAzureBlobName(blobId);
+      CloudBlockBlob azureBlob = azureContainer.getBlockBlobReference(azureBlobName);
+      azureBlob.setMetadata(getMetadataMap(cloudBlobMetadata));
+      azureBlob.upload(blobInputStream, inputLength, condition, options, blobOpContext);
+      logger.debug("Uploaded blob {} to Azure container {}.", blobId, azureContainer.getName());
+      return true;
+    } catch (StorageException sex) {
+      if (sex.getHttpStatusCode() == HttpURLConnection.HTTP_CONFLICT) {
+        // The blob already exists
+        logger.debug("Skipped upload of existing blob {}.", blobId);
+        azureMetrics.blobUploadSkippedCount.inc();
```

Comment:
Let's rename it to `blobUploadConflictCount` because there is a same name metric in vcrMetrics.blobUploadSkippedCount. They indicate different things.

Reply:
Done